src/main/java/org/opentripplanner/middleware/controllers/api/ApiKeyManagementController.java

```diff
@@ -0,0 +1,150 @@
+import org.opentripplanner.middleware.models.ApiUser;
+import org.opentripplanner.middleware.persistence.Persistence;
+import org.opentripplanner.middleware.persistence.TypedPersistence;
+import org.opentripplanner.middleware.utils.ApiGatewayUtils;
+import org.opentripplanner.middleware.utils.JsonUtils;
+import spark.Request;
+import spark.Response;
+import spark.Service;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.Collections;
+import java.util.List;
+
+import static org.opentripplanner.middleware.spark.Main.getConfigPropertyAsText;
+import static org.opentripplanner.middleware.utils.JsonUtils.logMessageAndHalt;
+
+/**
+ * Sets up HTTP endpoints for managing api keys and viewing api key request logs.
+ */
+public class ApiKeyManagementController {
+
+    private static TypedPersistence<ApiUser> apiUsers = Persistence.apiUsers;
+
+    private static final String DEFAULT_USAGE_PLAN_ID
+        = getConfigPropertyAsText("DEFAULT_USAGE_PLAN_ID");
+
+    /**
+     * Register http endpoints with {@link spark.Spark} instance at the provided API prefix.
+     */
+    public static void register(Service spark, String apiPrefix) {
+        // FIXME need to agree on endpoint (/admin or /secure or a mix) and endpoint names
+        spark.get(apiPrefix + "/admin/apikey/create", ApiKeyManagementController::createApiKeyForApiUser, JsonUtils::toJson);
+        spark.get(apiPrefix + "/admin/apikey/delete", ApiKeyManagementController::deleteApiKeyForApiUser, JsonUtils::toJson);
+        spark.get(apiPrefix + "/admin/apikey/usage", ApiKeyManagementController::getUsageLogsForApiUser, JsonUtils::toJson);
+    }
+
+    /**
+     * Create a new API key and assign it to the provided usage plan. If no usage plan is provided use the default
+     * usage plan instead.
+     */
+    private static CreateApiKeyResult createApiKeyForApiUser(Request req, Response res) {
+
+        ApiUser user = getApiUser(req);
+
+        String usagePlanId = req.queryParamOrDefault("usagePlanId", DEFAULT_USAGE_PLAN_ID);
+        // FIXME Should an Api user be limited to one api key per usage plan (and perhaps stage)?
+        CreateApiKeyResult apiKey = ApiGatewayUtils.createApiKey(user.id, usagePlanId);
+        if (apiKey == null) {
+            logMessageAndHalt(req,
+                HttpStatus.INTERNAL_SERVER_ERROR_500,
+                String.format("Unable to get AWS api key for user (%s)", user),
+                null);
+        }
+
+        // add new api key to user and persist
+        user.apiKeyIds.add(apiKey.getId());
+        apiUsers.replace(user.id, user);
+
+        // FIXME assumption is that the api key will be required by the calling service
+        return apiKey;
+    }
+
+    /**
+     * Delete an api key from a given user's list of api keys (if present) and from AWS api gateway.
+     */
+    private static boolean deleteApiKeyForApiUser(Request req, Response res) {
+
+        ApiUser user = getApiUser(req);
```

Comment:
We should allow an AdminUser to delete any key and an ApiUser to delete only their own keys.

Reply:
Updated.